ecosystem/models/configuration.py

```diff
@@ -53,39 +54,31 @@ def default_version(cls) -> List[str]:
         return ["3.6", "3.7", "3.8", "3.9"]
 
 
+@dataclass
 class RepositoryConfiguration(JsonSerializable):
-    """Configuration for ecosystem repository."""
-
-    def __init__(
-        self,
-        language: Optional[LanguageConfiguration] = None,
-        dependencies_files: Optional[List[str]] = None,
-        extra_dependencies: Optional[List[str]] = None,
-        tests_command: Optional[List[str]] = None,
-        styles_check_command: Optional[List[str]] = None,
-        coverages_check_command: Optional[List[str]] = None,
-    ):
-        """Configuration for ecosystem repository.
-
-        Args:
-            language: repository language configuration
-            dependencies_files: list of dependencies files paths relative to root of repo
-                ex: for python `requirements.txt`
-            extra_dependencies: list of extra dependencies for project to install during tests run
-                ex: for python it might be `qiskit==0.19`
-            tests_command: list of commands to run tests
-                ex: for python `python -m unittest -v`
-            styles_check_command: list of commands to run style checks
-                ex: for python `pylint -rn ecosystem tests`
-            coverages_check_command: list of commands to run coverage checks
-                ex: for python `coverage3 -m unittest -v && coverage report`
-        """
-        self.language = language or PythonLanguageConfiguration()
-        self.dependencies_files = dependencies_files or []
-        self.extra_dependencies = extra_dependencies or []
-        self.tests_command = tests_command or []
-        self.styles_check_command = styles_check_command or []
-        self.coverages_check_command = coverages_check_command or []
+    """
+    Configuration for ecosystem repository.
+
+    Attributes:
+        language: repository language configuration
+        dependencies_files: list of dependencies files paths relative to root of repo
+            ex: for python `requirements.txt`
+        extra_dependencies: list of extra dependencies for project to install during tests run
+            ex: for python it might be `qiskit==0.19`
+        tests_command: list of commands to run tests
+            ex: for python `python -m unittest -v`
+        styles_check_command: list of commands to run style checks
+            ex: for python `pylint -rn ecosystem tests`
+        coverages_check_command: list of commands to run coverage checks
+            ex: for python `coverage3 -m unittest -v && coverage report`
+    """
+
+    language: LanguageConfiguration = field(default_factory=PythonLanguageConfiguration)
+    dependencies_files: List[str] = new_list()
```

Comment:
If you add `from __future__ import annotations` to the top of the file, you can rewrite this to `list[str]`. No need to import `List`. Much more ergonomic.

Reply:
Nice! Fixed in [7056e8a](https://github.com/qiskit-community/ecosystem/pull/457/commits/7056e8ae0f4bc834bc9ba094faa374c1a8d56d89)